src/components/ListView/SocialArea.jsx

```diff
@@ -0,0 +1,107 @@
+import { useState, useCallback } from 'react';
+import { FiThumbsUp, FiShare2, FiHeart } from 'react-icons/fi';
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import Message from '../Message';
+
+function SocialArea({ likes }) {
+  const [likeArr, setlikeArr] = useState([likes, undefined]);
+  const [heartArr, setHeartArr] = useState([false, undefined]);
+
+  const likeClick = useCallback(() => {
+    const changeToNum = parseInt(likeArr[0], 10);
+    if (changeToNum === parseInt(likes, 10)) {
+      setlikeArr([changeToNum + 1, 'black']);
+    } else {
+      setlikeArr([changeToNum - 1, undefined]);
+    }
+  }, [likeArr, likes]);
+
+  const heartClick = useCallback(() => {
+    if (heartArr[0]) {
+      setHeartArr([!heartArr[0], 'black']);
+    } else {
+      setHeartArr([!heartArr[0], undefined]);
+    }
```

Comment:
```suggestion
      setHeartArr([!heartArr[0], heartArr[0] ? 'black' : undefined]);
```
**Recommend-Request**: 축약이 가능할 것 같습니다!